File: src/main/java/com/github/tianchenghang/model/vo/AppVo.java
package com.github.tianchenghang.model.vo;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;
import lombok.Data;

@Data
public class AppVo implements Serializable {

  private Long id;

  private String appName;

  private String appCover;

  private String initPrompt;

  private String codegenType;

  private String deployKey;

  private LocalDateTime deployTime;

  private Integer priority;

  private Long userId;

  private LocalDateTime createTime;

  private LocalDateTime updateTime;

  private UserVo user;

  @Serial private static final long serialVersionUID = 1L;
}

File: apps/server/src/app-module/vo/app-vo.ts
import { AppEntity } from '../../database/entities/app-entity';

export class AppVo {
  id = '';
  appName = '';
  appCover = '';
  initPrompt = '';
  codegenType = '';
  deployKey = '';
  deployTime = '';
  priority = 0;
  userId = '';
  createTime = new Date();
  updateTime = new Date();

  static fromEntity(entity: AppEntity): AppVo | null {
    if (!entity) return null;
    const vo = new AppVo();
    vo.id = String(entity.id);
    vo.appName = entity.appName;
    vo.appCover = entity.appCover;
    vo.initPrompt = entity.initPrompt;
    vo.codegenType = entity.codegenType;
    vo.deployKey = entity.deployKey;
    vo.deployTime = entity.deployTime;
    vo.priority = entity.priority;
    vo.userId = String(entity.userId);
    vo.createTime = entity.createTime;
    vo.updateTime = entity.updateTime;
    return vo;
  }
}
